### PR TITLE
add functional tests

### PR DESCRIPTION
**Pull request by cvan**

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  var peer2 = new SocketPeer({
+    pairCode: 'yolo',
+    url: 'http://localhost:3000',
+    reconnect: false,
+  });
+
+  peer1.on('connect', function () {
+    t.equal(peer1.socketConnected, true);
+    t.equal(peer1._connections.socket.attempt, 0);
```
Comment (cvan) [inline test/index.js]: add assertion messages for these

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    url: 'http://localhost:3000'
+  });
+
+  peer.on('connect_attempt', function () {
+    t.pass('got connect_attempt event');
+    t.equal(peer.socketConnected, false);
+    t.equal(peer._connections.socket.attempt, 1);
+    t.equal(peer._connections.socket.success, 0);
+    t.equal(peer._connections.socket.error, 0);
+    peer.destroy();
```
Comment (cvan) [inline test/index.js:20]: I see this JS error:

> Uncaught TypeError: Uncaught, unspecified "error" event.events.js:61 EventEmitter.emitindex.js:9 SocketPeer._socketErrorindex.js:9 SocketPeer.closeindex.js:9 SocketPeer.destroyindex.js:18 (anonymous function)events.js:74 EventEmitter.emitindex.js:9 SocketPeer.connectindex.js:9 (anonymous function)

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    t.equal(peer._connections.socket.error, 1, 'connect_timeout.error');
+    peer.destroy();
+    t.end();
+  });
+
+});
+
+
+
+test('connect_timeout event gets emitted', function (t) {
```
Comment (cvan) [inline test/index.js]: this is duplicated

File: browser/index.js [hunk 4]
```diff
@@ -118,7 +116,9 @@ SocketPeer.prototype.connect = function () {
   self.socket.onopen = function () {
     self.pair();
   };
-  self.socket.onerror = self._socketError.bind(self);
+  self.socket.onerror = function (event) {
```
Comment (cvan) [inline browser/index.js:119]: notice this

File: .zuul.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+ui: tape
+browsers:
```
Comment (potch) [inline .zuul.yml]: are we using sauce yet? if not we don't need the browser list.

File: browser/index.js (hunk 4 above)
Comment (potch) [inline browser/index.js:119]: is this so it's wrapped in a proper `Error` object?

File: .zuul.yml (hunk 5 above)
Comment (cvan) [inline .zuul.yml]: okay, removed

File: browser/index.js (hunk 4 above)
Comment (cvan) [inline browser/index.js:119]: no, `onerror` returns an `Event` not an `Error` object. `event.data` is the `Error` object (if there even is one [P.S. there's not always one]).

**Comment by cvan** [comment]
# r+

**Pull request merged**